Feature/summarize

AKallala opened the pull request:
This PR implements the summarize file tool of the pdf converter, it:
- Makes the file picker launch on summarize file option click on the `PdfConverterScreen` for the user to select a pdf file to summarize from local storage
- Implements in `PdfConverterViewModel` the logic to generate a pdf file containing the summary of the input file
- Adds a function to `PdfRepository` to read the text from a pdf file
- Adds a function to `PdfRepository` to write a string to a `PdfDocument`
- Updates summarizeText function in `OpenAiRepository` to use openAi's gpt-3.5-turbo model instead of text-davinci-003 model which is deprecated 

### Ideas for future improvements:
- Make the summarization tool handle bigger input files: for now the summarization is made through a single request to openAi's api and the latter can handle at most 4096 tokens (input + output text) per request which limits the maximum size of the text that can be summarized
- Fine tune the summarization by generating better summaries given the kind of input file. Because this first version summarizes all inputs by sending the same prompt to the api, which leads to poor summaries regarding the nature of the input file. The idea here is to adapt the prompts to each case in order to get better summaries (for example in the case of a lecture file the summary should be structured in a way that highlights the key concepts of the lecture, in the case of a story just provide a concise summary of the latter)

MathieuxC reviewed (changes requested):
Your code seems good, but I don't understand why you changed the ui of PdfConverterScreen. If you just created your branch before #197, merge main in your branch before going the other way around.
Otherwise lgtm, juste change this and I will approve

AKallala commented:
Sorry the branch was behind the main branch and I didn't pay attention when doing the merge and forgot to keep the changes made on PdfCanverterScreen, I'm going to push the correct version

AKallala commented on app/src/main/java/com/github/se/eduverse/ui/converter/PdfConverter.kt:
```diff
@@ -249,10 +244,8 @@ fun OptionCard(
-              contentColor = MaterialTheme.colorScheme.onPrimaryContainer)) {
+      elevation = CardDefaults.cardElevation(8.dp),
+      colors = CardDefaults.cardColors(containerColor = Color(0xFFC2F5F0))) {
         Column(
```
An error in the merge made me modify the old version of the file instead of the latest version, I will rectify it

AKallala replied:
Done

AKallala commented on app/src/main/java/com/github/se/eduverse/viewmodel/PdfConverterViewModel.kt:
```diff
@@ -142,4 +156,24 @@ class PdfConverterViewModel(private val pdfRepository: PdfRepository) : ViewMode
+          Log.e("getSummary", "Failed to get summary from openAi api", it)
+          throw it
+        })
+  }
```
Sorry but I don't understand what you're suggesting, can you explain

AKallala commented on app/src/main/java/com/github/se/eduverse/ui/converter/PdfConverter.kt:
```diff
@@ -76,9 +74,6 @@ fun PdfConverterScreen(
         TopNavigationBar(screenTitle = "PDF Converter", navigationActions = navigationActions)
-      },
-      bottomBar = {
-        BottomNavigationMenu({ navigationActions.navigateTo(it) }, LIST_TOP_LEVEL_DESTINATION, "")
```
The removal was made by mistake, i fixed it

AKallala commented:
The reason I reverted #198 is that I created by mistake a pull request(#199) before pushing my committed changes to the branch. Which lead to a third check (CI (pull_request)) being launched on the main branch, and the only way I found to get rid of this check without deleting the branch which already had my latest changes committed was to revert the latest commit made to the main branch namely the merge #198 and I added these changes to this branch
